src/NumericalAlgorithms/Interpolation/InterpolationTargetApparentHorizon.hpp

```diff
@@ -0,0 +1,182 @@
+/// \endcond
+
+namespace intrp {
+
+namespace OptionHolders {
+/// Options for finding an apparent horizon.
+template <typename Frame>
+struct ApparentHorizon {
+  struct InitialGuess {
+    static constexpr OptionString help = {"Initial guess"};
+    using type = Strahlkorper<Frame>;
+  };
+  struct FastFlow {
+    static constexpr OptionString help = {"FastFlow options"};
+    using type = ::FastFlow;
+  };
+  using options = tmpl::list<InitialGuess, FastFlow, ::Tags::Verbosity>;
+  static constexpr OptionString help = {
+      "A Strahlkorper and FastFlow used for horizon finding."};
+
+  ApparentHorizon(Strahlkorper<Frame>&& strahlkorper_in,
+                  ::FastFlow&& fast_flow_in,
+                  Verbosity && verbosity_in) noexcept;
+
+  ApparentHorizon() = default;
+  ApparentHorizon(const ApparentHorizon& /*rhs*/) = default;
+  ApparentHorizon& operator=(const ApparentHorizon& /*rhs*/) = default;
+  ApparentHorizon(ApparentHorizon&& /*rhs*/) noexcept = default;
+  ApparentHorizon& operator=(ApparentHorizon&& /*rhs*/) noexcept = default;
+  ~ApparentHorizon() = default;
+
+  // clang-tidy non-const reference pointer.
+  void pup(PUP::er& p) noexcept;  // NOLINT
+
+  Strahlkorper<Frame> strahlkorper{};
+  ::FastFlow fast_flow{};
+  Verbosity verbosity{Verbosity::Quiet};
+};
+
+template <typename Frame>
+bool operator==(const ApparentHorizon<Frame>& lhs,
+                const ApparentHorizon<Frame>& rhs) noexcept;
+template <typename Frame>
+bool operator!=(const ApparentHorizon<Frame>& lhs,
+                const ApparentHorizon<Frame>& rhs) noexcept;
+
+}  // namespace OptionHolders
+
+namespace Actions {
+/// \ingroup ActionsGroup
+/// \brief Sends points on a trial apparent horizon to an `Interpolator`.
+///
+/// This differs from `KerrHorizon` in the following ways:
+/// - It supplies points on a prolonged Strahlkorper, at a higher resolution
+///   than the Strahlkorper in the DataBox, as needed for horizon finding.
+/// - It uses a `FastFlow` in the DataBox.
+/// - It has different options (including those for `FastFlow`).
+///
+/// Uses:
+/// - DataBox:
+///   - `::Tags::Domain<VolumeDim, Frame>`
+///   - `::Tags::FastFlow`
+///   - `StrahlkorperTags::CartesianCoords<Frame>`
+///   - `::Tags::Variables<typename
+///                   InterpolationTargetTag::vars_to_interpolate_to_target>`
+///
+/// DataBox changes:
+/// - Adds: nothing
+/// - Removes: nothing
+/// - Modifies:
+///   - `Tags::IndicesOfFilledInterpPoints`
+///   - `::Tags::Variables<typename
+///                   InterpolationTargetTag::vars_to_interpolate_to_target>`
+///
+/// This Action also has an initialize function that adds to the DataBox:
+/// - `StrahlkorperTags::items_tags<Frame>`
+/// - `StrahlkorperTags::compute_items_tags<Frame>`
+/// - `::Tags::FastFlow`
+/// - `::Tags::Verbosity`
+///
+/// For requirements on InterpolationTargetTag, see InterpolationTarget
+template <typename InterpolationTargetTag, typename Frame>
+struct ApparentHorizon {
```

Comment:
This Action doesn't actually find the ApparentHorizon.  It sends points on a trial apparent horizon to an Interpolator.  Other similar actions are called, e.g. `KerrHorizon` and `LineSegment`, which sends points on an (analytic) Kerr horizon or on a specified LineSegment to an Interpolator.

I guess all the similar actions could be renamed `SendPointsOnKerrHorizonToInterpolator`, `SendPointsOnLineSegmentToInterpolator`, `SendPointsOnTrialApparentHorizonToInterpolator`, `SendPointsOnWedgeSectionTorusToInterpolator`, which are more descriptive, but those names seem to me very verbose.

Reply:
If I did such a name change, I'd like to do it in a separate PR which would be easy to review since it's only a name change.

Reply:
If I understand correctly, this action does kick off the apparent horizon finding by sending points to the interpolator, which then calls the callback. So in the action list of the elements I think that should be clear.

I agree you can follow the naming patter of other similar actions and do a rename later.

Reply:
(OK.  By the way, this Action will not appear in the action list of the elements.  It will appear in a type alias inside an InterpolationTarget specified in the Metavariables, as in the test.)

Reply:
This is indeed an Action, not a callback.  But it's not an Action that is executed by an Element component.  It is an Action that is executed by an InterpolationTarget component.  And it is a "simple action", not an "iterable action", so it doesn't appear in an action list.

Reply:
Ok I think it would be very helpful if you could add an overview of how the different components of the AH finder and the interpolation framework work together, since I found this a bit hard to understand from reading the individual docs and the tests. A great place for this would be an `ApparentHorizonGroup` Doxygen group, where you could also collect the relevant types and functions in one place. This would help me a lot, and hopefully also others.

Reply:
I just made issue #1326

Reply:
Ok sounds good